addons/misra.py

```diff
@@ -1611,6 +1611,14 @@ def misra_17_8(self, data):
             if var and var.isArgument:
                 self.reportError(token, 17, 8)
 
+    def misra_18_4(self, data):
+        for token in data.tokenlist:
+            if not token.str in {'+', '-', '+=', '-='}:
+                continue
+            var1 = token.astOperand1.variable
```

Comment:
Please use the `valueType` instead of `variable`. Please check that the `astOperand1` and `astOperand2` are not None.

Reply:
In which case we could have `None` value in astOperand's?

Reply:
if the code is:

    x = -y;

the `-` will only have astOperand1.

But otherwise the astOperand1 _should_ be set I guess. In declarations the ast is not created so if the code is `int * p;` then the `*` will not have operands. 

Maybe some code like this would mean that there is no operand:

    UNKNOWN_MACRO(+)

But well.. we might write a Syntax Error in some of those cases.

Reply:
Cppcheck does not have strong syntax checking by intention.